example/eventGenerator/EventTelemetryProvider.js

```diff
@@ -0,0 +1,93 @@
+/*****************************************************************************
+ * Open MCT, Copyright (c) 2014-2021, United States Government
+ * as represented by the Administrator of the National Aeronautics and Space
+ * Administration. All rights reserved.
+ *
+ * Open MCT is licensed under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * Open MCT includes source code licensed under additional open source
+ * licenses. See the Open Source Licenses file (LICENSES.md) included with
+ * this source code distribution or the Licensing information page available
+ * at runtime from the About dialog for additional information.
+ *****************************************************************************/
+
+/**
+ * Module defining EventTelemetryProvider. Created by chacskaylo on 06/18/2015.
+ */
+
+import messages from './transcript.json';
+
+class EventTelemetryProvider {
+    constructor() {
+        this.firstObservedTime = Date.now();
+        this.count = 0;
+        this.defaultSize = 25;
+    }
+
+    generateData(startTime, duration, name) {
+        const millisecondsSinceStart = startTime - this.firstObservedTime;
+        const utc = Math.floor(startTime / duration) * duration;
+        const ind = this.count % messages.length;
+        const message = messages[ind] + " - [" + millisecondsSinceStart + "]";
+        this.count += 1;
```

Comment:
This will cause a really subtle bug that may manifest as the transcript skipping lines. In all likelihood no one would even notice :) It's worth pointing out though because it crops up from time to time in other telemetry providers.

Basically, telemetry providers shouldn't track any state on member variables. The problem is that the Telemetry API will store just a single instance of each telemetry provider (as opposed to using a new instance per subscription or telemetry request). There can be multiple simultaneous subscriptions and requests for telemetry, so it's important not to store any subscription or request-specific state on the telemetry providers themselves, since the instance will be shared.

So, in this case the count will need to be tracked in the subscribe and request functions themselves, and then passed down to `generateData`. It's a little ugly, but it will avoid this issue.

Reply:
easy enough, and thanks for noticing it! you're correct in that I wouldn't have found it!